src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,376 @@
+               
+                String sig = "";
+                String returnT = "";
+                
+                if (member.tagname.equals("method")) {
+                    if (member.parameters.size() > 0) {
+                        doc += "<br>"
+                            + "<h3 class=\"pa\">Parameters</h3>"
+                            + "<ul>";
+                    }
+                    
+                    sig = "(";
+
+                    for (ParameterObject parameter : member.parameters) {
+                        
+                        //Are we a known class being passed in? if so render a link to class
+                        String datatype;
+                        if(classes.get(parameter.datatype) != null)
+                          datatype = renderLink(parameter.datatype);
+                        else 
+                          datatype = parameter.datatype;
+                        
+                        if (sig.equals("("))
+                            sig += datatype;
+                        else
+                            sig += ", " + datatype;
+                        
+                        doc += renderParams(parameter);
+                    }
+                    sig += ")";
+                    
+                    if (member.parameters.size() > 0) {
+                        doc += "</ul>";
+                    }
+                    
+                    Matcher r = returnType.matcher(member.signature);
+                    if(r.find()) {
+                        if(classes.get(r.group(1)) != null){
+                            returnT = renderLink(r.group(1));
+                        } else {
+                            returnT = r.group(1);
+                        }
+                    }
+                    
+                    doc += renderReturns(returnT, member.returnComment);
+                }
+                
+                if (member.tagname.equals("property")) {
+                    if(classes.get(member.datatype) != null){
+                        returnT = renderLink(member.datatype);
+                    } else {
+                        returnT = member.datatype;
+                    }
+                }
+                
+                memberHTML += "<div id='" + member.id + "' class='member " + firstChild + " " + inherited + "'>"
+                              // leftmost column: expand button
+                              + "<a href='#' class='side expandable'>"
+                                  + "<span>&nbsp;</span>"
+                              + "</a>"
+                              // member name and type + link to owner class and source
+                              + "<div class='title'>"
+                                  + "<div class='meta'>"
+                                      + "<span class='defined-in' rel='" + member.owner + "'>" + member.owner + "</span>"

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "</span>" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1192)